### PR TITLE
Feat/fly 2544

### PR DESCRIPTION
**Pull request by Hakob23**
## What

Implements `claimPegOut` on PegOutEscrow: an LP becomes responsible for delivery and the user's escrowed RBTC moves into PegOutContract in one transaction.

Adds a minimal settlement entry `registerClaimedPegOut` on PegOutContract (escrow-only), plus AC coverage in `test/pegout-escrow/PegOutEscrow.t.sol`.

Stacked on `feat/FLY-2543`.

## Why

Commit-first peg-out discovery is global, so claim is a race. The first check is state (`REQUESTED`), so a losing LP pays for a cheap revert. Claim is a hard commitment (no unclaim): collateral + signature bind the LP before funds leave escrow, and transfer + responsibility record must land atomically.

## Type of Change

- [ ] Bug fix (non-breaking change)
- [x] New feature (non-breaking change)
- [ ] Breaking change
- [ ] Refactor (no intended behavior change)
- [ ] Security hardening
- [ ] Tests only
- [ ] Build/CI/Tooling
- [ ] Documentation

## Affected Areas

- [ ] PegIn flow (`registerPegIn`, `callForUser`, quote validation)
- [x] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)
- [ ] Liquidity Provider lifecycle (register/update/status/resign/collateral/withdraw)
- [ ] Pause/operational controls
- [x] Quote hashing/signature logic
- [ ] Bitcoin tx / PMT / merkle validation
- [ ] Deployment or upgrade scripts / Makefile targets
- [ ] CI workflows (`ci`, `fuzz`, `formal`, `slither`, `codeql`)
- [ ] Docs

## Risk & Security Review

- [x] Access control and authorization paths reviewed
- [x] Reentrancy and external call paths reviewed
- [x] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [x] Time/block/confirmation assumptions reviewed
- [ ] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

Notes:
- `PegOutContract` gains `_pegOutEscrow` storage before existing mappings (new wiring on this stack; review upgrade layout if this lands on already-deployed PegOut proxies).
- Legacy `depositPegOut` is left enabled (no `LegacyDepositDisabled` in this PR).

## Test Plan

List what you ran locally and the outcome:

- [ ] `npm run lint:sol`
- [ ] `npm run lint:ts`
- [ ] `npm run compile`
- [ ] `npm test`
- [ ] `npm run test:fuzz` (if logic/state transitions changed)
- [ ] `npm run test:invariant` (if invariant-sensitive logic changed)
- [ ] `npm run test:formal` (if formal-verification-sensitive logic changed)
- [ ] `npm run test:integration` (if integration paths changed)
- [ ] `npm run test:e2e` (if deployment/ownership flows changed)

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

- Local: `forge test --match-path 'test/pegout-escrow/*'` (18 passed) and `forge test --match-path 'test/pegout/*'` during implementation.
- AC coverage for `claimPegOut`: second claim, claim after cancel, claim after refund (state forced to `REFUNDED` via storage write — `refundOnNoClaim` not in this PR), not registered / under-collateralized (`vm.mockCall` on `CollateralManagementMock`), bad signature, happy path (LP, funds, `PegOutDeposit` timestamp).
- `refundOnNoClaim` / `onSettlement` remain stubbed.

## Deployment & Ops Impact

- [ ] No deployment impact
- [x] Requires deployment
- [x] Requires upgrade
- [ ] Env/config changes required (`.env`, network RPCs, verification, etc.)
- [ ] Runbook/update instructions added to PR description

Requires PegOutEscrow + PegOutContract upgrades and admin `setPegOutEscrow(escrow)` on PegOutContract before claims can succeed.

## Related Issues

- Jira: [FLY-2544](https://rsklabs.atlassian.net/browse/FLY-2544)

## Reviewer Notes

**Claim order (escrow):** `REQUESTED` → registered → collateral sufficient → claim window → set `lpRskAddress` + EIP-712 verify → `CLAIMED` + `PegOutClaimed` → `registerClaimedPegOut{value}`.

**Minimal PegOut entry:** escrow-only `registerClaimedPegOut`; quote body stays on escrow; registry stores claim timestamp/block; emits `PegOutDeposit`.

**Intentional product notes (call out in review):**
1. **No public claim-anchor getter.** Anchors live in PegOutContract’s private registry. Claim timestamp is observable via `PegOutDeposit`; a view for block/timestamp can be added later if LPS/SDK need an on-chain read.
2. **Collateral gate split on the claim path** (`claimPegOut` and `registerClaimedPegOut`): `isRegistered` → `ProviderNotRegistered`, then `isCollateralSufficient` → `InsufficientCollateral`. Legacy `depositPegOut` still maps insufficient collateral to `ProviderNotRegistered`.
3. **`QuoteAlreadyCompleted` vs `QuoteAlreadyRegistered`:** `depositTimestamp` is not cleared on settle, so the registered check alone would also block completed ids; `completed` is checked first for a precise post-settlement revert (comment in code).
4. Out of scope: `refundOnNoClaim`, `onSettlement`, global slash, disabling legacy deposit, claim-fail discipline.

### CODE REVIEW

File: src/interfaces/IPegOut.sol
Comment (Luisfc68) [inline src/interfaces/IPegOut.sol:1]: the idea was to maintain `depositPegOut` function, notice the one you added does almost the same, I recommend to maintain `depositPegOut` and just add two new checks, one to ensure the caller is the escrow and the other one to check the quote was claimed (the one you already have)

File: src/PegOutContract.sol
Comment (Luisfc68) [inline src/PegOutContract.sol:1]: The pegout contract should not be aware of the concept of "request", that's the escrow's responsibility, by this point the quote is already claimed so there's no need to introduce that here, you can still use the same concepts that we've been using in the pegout contract

File: src/PegOutContract.sol [hunk 1]
```diff
@@ -331,12 +396,25 @@ contract PegOutContract is
...
+        quote = _pegOutEscrow.getPegOutQuote(requestHash);
+    }
+
     /// @notice This function is used to hash a peg out quote
     /// @dev The function also validates the quote belongs to this contract
     /// @param quote the peg out quote to hash
     /// @return quoteHash the hash of the peg out quote
     function _hashPegOutQuote(
-        Quotes.PegOutQuote calldata quote
+        Quotes.PegOutQuote memory quote
```
Comment (Luisfc68) [inline src/PegOutContract.sol:417]: why changing this? calldata is more gas efficient

File: src/PegOutEscrow.sol [hunk 2]
```diff
@@ -172,9 +173,42 @@ contract PegOutEscrow is
...
+        if (!SignatureValidator.verify(msg.sender, eip712Hash, signature)) {
+            revert SignatureValidator.IncorrectSignature(msg.sender, eip712Hash, signature);
+        }
+
+        $.state[requestHash] = EscrowedPegOutState.CLAIMED;
+
+        uint256 valueToSend = q.value + q.callFee + q.gasFee;
+        emit PegOutClaimed(msg.sender, requestHash);
+
+        $.pegOutContract.registerClaimedPegOut{value: valueToSend}(requestHash, signature);
```
Comment (Luisfc68) [inline src/PegOutEscrow.sol:211]: You'll need to change a lot of stuff unnecesarily because of this, my recommendation is that you expect the quote hash + the signature, the LP computes it by taking the quote and completing it with its address and then signing that. Thats why I recommended using the incomplete quote as request hash, if you go that way, the PegOutContract needs almost no changes

**Review by Luisfc68 — CHANGES_REQUESTED** [review]

**Comment by Hakob23** [comment]
Fixes are applied in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

File: src/interfaces/IPegOut.sol
Comment (Hakob23) [reply src/interfaces/IPegOut.sol]: Fixed in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

File: src/PegOutContract.sol
Comment (Hakob23) [reply src/PegOutContract.sol]: Fixed in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

File: src/PegOutContract.sol (hunk 1 above)
Comment (Hakob23) [reply src/PegOutContract.sol]: Fixed in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

File: src/PegOutEscrow.sol (hunk 2 above)
Comment (Hakob23) [reply src/PegOutEscrow.sol]: Fixed in https://github.com/rsksmart/liquidity-bridge-contract/pull/549
